temporal-sdk/src/test/java/io/temporal/workflow/DynamicWorkflowTest.java

```diff
@@ -119,4 +126,26 @@ public void testDynamicWorkflowFactory() {
     String result = workflow.getResult(String.class);
     assertEquals("activityType2-activityType1-startArg0-workflowFoo", result);
   }
+
+  @Test
+  public void testDynamicWorkflowFailure() {
+    TestWorkflowEnvironment testEnvironment = testWorkflowRule.getTestEnvironment();
+    testEnvironment
+        .getWorkerFactory()
+        .getWorker(testWorkflowRule.getTaskQueue())
+        .registerWorkflowImplementationTypes(DynamicWorkflowImpl.class);
+    testEnvironment.start();
+
+    WorkflowOptions workflowOptions =
+        WorkflowOptions.newBuilder().setTaskQueue(testWorkflowRule.getTaskQueue()).build();
+    WorkflowStub workflow =
+        testWorkflowRule.getWorkflowClient().newUntypedWorkflowStub("workflowFoo", workflowOptions);
+    workflow.start("startArg0", true /* fail */);
+    try {
+      workflow.getResult(String.class);
+      fail("failure expected");
+    } catch (WorkflowFailedException e) {
+      // expected
```

Comment:
To make it more clear and explicit, it's a good idea to don't catch the exception, add WorkflowFailedException to the test method signature and add it to @Test(expected = WorkflowFailedException.class)